File: app/src/main/java/br/com/accenture/santander/wallacebaldenebre/ui/adapter/StatementsAdapter.java
package br.com.accenture.santander.wallacebaldenebre.ui.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import br.com.accenture.santander.wallacebaldenebre.R;
import br.com.accenture.santander.wallacebaldenebre.model.StatementList;

import java.text.NumberFormat;
import java.util.ArrayList;

public class StatementsAdapter extends RecyclerView.Adapter<StatementsAdapter.ViewHolder> {
    private ArrayList<StatementList> list;

    public StatementsAdapter(ArrayList<StatementList> list) {
        this.list = list;
    }

    @Override
    public StatementsAdapter.ViewHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
        return new ViewHolder(LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_bill, viewGroup, false));
    }

    @Override
    public void onBindViewHolder(StatementsAdapter.ViewHolder holder, int position) {
        holder.title.setText(list.get(position).getTitle());
        holder.type.setText(list.get(position).getDesc());
        holder.date.setText(list.get(position).getDate());
        holder.value.setText(NumberFormat.getCurrencyInstance().format(list.get(position).getValue()) + "");
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        public TextView title;
        public TextView type;
        public TextView date;
        public TextView value;

        public ViewHolder(View v) {
            super(v);
            title = v.findViewById(R.id.tvw_item_bill_title);
            type = v.findViewById(R.id.tvw_item_bill_type);
            date = v.findViewById(R.id.tvw_item_bill_date);
            value = v.findViewById(R.id.tvw_item_bill_price);
        }
    }
}

File: app/src/main/java/com/example/testesantander/ui/statements/StatementsAdapter.kt
package com.example.testesantander.ui.statements

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.testesantander.R
import com.example.testesantander.domain.model.StatementsData
import com.example.testesantander.utils.MoneyUtil
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.adapter_statements.*
import java.text.DecimalFormat
import java.text.NumberFormat
import java.text.SimpleDateFormat
import java.util.*

class StatementsAdapter(private val statementsList: Array<StatementsData>): RecyclerView.Adapter<StatementsAdapter.ViewHolder>(){

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val context = parent.context
        val view = LayoutInflater.from(context).inflate(R.layout.adapter_statements, parent, false)
        return ViewHolder(view)
    }

    override fun getItemCount(): Int {
        return statementsList.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val context = holder.tvDate.context
        val sdf = SimpleDateFormat("yyyy-MM-dd")
        val date = sdf.parse(statementsList[position].date)
        sdf.applyPattern("dd/MM/yyyy")
        val value = statementsList[position].value



        holder.tvTitle.text = statementsList[position].title
        holder.tvDesc.text = statementsList[position].desc
        holder.tvDate.text = sdf.format(date).toString()
        holder.tvValue.text = MoneyUtil.moneyPtBr(value)
    }


    class ViewHolder(override val containerView: View): RecyclerView.ViewHolder(containerView), LayoutContainer
}